### PR TITLE
Remove SHOUTY banners

### LINKED ISSUE #306: STL: WE SHOULD REMOVE THE SHOUTY COMMENT BANNERS!
We currently follow a convention of introducing structs, classes, functions, etc. with "SHOUTY COMMENT BANNERS". A few examples:

https://github.com/microsoft/STL/blob/580e61a5f5536016f521cc1b2ca636eadc69bd30/stl/inc/vector#L410-L412

https://github.com/microsoft/STL/blob/580e61a5f5536016f521cc1b2ca636eadc69bd30/stl/inc/algorithm#L29-L41

https://github.com/microsoft/STL/blob/580e61a5f5536016f521cc1b2ca636eadc69bd30/stl/inc/type_traits#L536-L541

We've been thinking about changing this convention. It hasn't been applied rigorously, even for publicly-documented components (e.g. we have `// STRUCT TEMPLATE is_empty` but not `// VARIABLE TEMPLATE is_empty_v`). It might have helped file navigation many years ago, but IDE/editor support for Go To Definition and Find In Files (e.g. VSCode Ctrl+Shift+F) makes it easy to find where components are defined.

In theory, the banners could be most useful when publicly-documented components are preceded by internal support machinery, but in practice, we haven't always used them like that. For example, some but not all of the helpers for `is_permutation()` are introduced with `// FUNCTION TEMPLATE is_permutation`:

https://github.com/microsoft/STL/blob/580e61a5f5536016f521cc1b2ca636eadc69bd30/stl/inc/xutility#L4395-L4455

As these comments seem to provide minimal value today, are commonly overlooked by new contributors, and consume valuable lines, we should decide whether to keep or discard them.

### PR DESCRIPTION
**Pull request by SuperWig**
Fixes #306

Not fully done yet but this should be all of the major ones. 

Should some remain like this one?

https://github.com/microsoft/STL/blob/bd7adb4a932725f60ba096580c415616486ab64c/stl/inc/atomic#L105-L109

P.S. Good luck with the review. 🙏

### CODE REVIEW

File: stl/src/wlocale.cpp [hunk 1]
```diff
@@ -1,7 +1,6 @@
 // Copyright (c) Microsoft Corporation.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-// class locale wide member functions
```
Comment (AlexGuteniev) [inline stl/src/wlocale.cpp:4]: I'm not sure if this is SHOUTY or useless

File: stl/src/xlocale.cpp [hunk 2]
```diff
@@ -1,7 +1,6 @@
 // Copyright (c) Microsoft Corporation.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-// class locale exotic char member functions
```
Comment (AlexGuteniev) [inline stl/src/xlocale.cpp:4]: I'm not sure if this is SHOUTY or useless

File: stl/src/wlocale.cpp (hunk 1 above)
Comment (SuperWig) [reply stl/src/wlocale.cpp:4]: Urgh, not sure how that happened.

File: stl/inc/set [hunk 3]
```diff
@@ -21,12 +21,10 @@ _STL_DISABLE_CLANG_WARNINGS
 #pragma push_macro("new")
 #undef new
 _STD_BEGIN
-// CLASS TEMPLATE _Tset_traits
-template <class _Kty, // key/value type
```
Comment (AlexGuteniev) [inline stl/inc/set:25]: What>

File: stl/inc/set [hunk 4]
```diff
@@ -49,114 +47,111 @@ public:
...
-    using const_reference        = const value_type&;
-    using iterator               = typename _Mybase::iterator;
-    using const_iterator         = typename _Mybase::const_iterator;
-    using reverse_iterator       = typename _Mybase::reverse_iterator;
-    using const_reverse_iterator = typename _Mybase::const_reverse_iterator;
-
-    using _Alnode        = typename _Mybase::_Alnode;
-    using _Alnode_traits = typename _Mybase::_Alnode_traits;
+// ordered red-black tree of key values, unique keys
+public : static_assert(!_ENFORCE_MATCHING_ALLOCATORS || is_same_v<_Kty, typename _Alloc::value_type>,
```
Comment (AlexGuteniev) [inline stl/inc/set:52]: This does not make sense

File: stl/src/locale.cpp [hunk 5]
```diff
@@ -1,7 +1,6 @@
 // Copyright (c) Microsoft Corporation.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-// class locale member functions
```
Comment (AlexGuteniev) [inline stl/src/locale.cpp:4]: Same, not sure if this should be removed

File: stl/src/locale0.cpp [hunk 6]
```diff
@@ -1,7 +1,6 @@
 // Copyright (c) Microsoft Corporation.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-// class locale basic member functions
```
Comment (AlexGuteniev) [inline stl/src/locale0.cpp:4]: Same, not sure if this should be removed

File: stl/inc/set (hunk 4 above)
Comment (SuperWig) [reply stl/inc/set:52]: I think VS Code might not have liked replacing so many....

(to note, I did audit the previews for each file before replacing)

File: stl/inc/functional [hunk 7]
```diff
@@ -29,16 +29,12 @@ _STL_DISABLE_CLANG_WARNINGS
...
 
 _STD_BEGIN
-// STRUCT TEMPLATE plus
 // defined in <xstddef>
 
-// STRUCT TEMPLATE minus
 // defined in <xstddef>
 
-// STRUCT TEMPLATE multiplies
 // defined in <xstddef>
```
Comment (SuperWig) [inline stl/inc/functional:36]: Should these be replaced with a comment saying the others are defined elsewhere?

File: stl/inc/functional [hunk 8]
```diff
@@ -73,25 +62,18 @@ struct negate {
...
 
-// STRUCT TEMPLATE less
 // defined in <xstddef>
 
-// STRUCT TEMPLATE greater_equal
 // defined in <xstddef>
 
-// STRUCT TEMPLATE less_equal
 // defined in <xstddef>
```
Comment (AlexGuteniev) [inline stl/inc/functional:76]: more to  fix

File: stl/inc/set (hunk 3 above)
Comment (SuperWig) [reply stl/inc/set:25]: Lesson learned I guess. Look at what VS Code actually replaces with 😅.

File: stl/inc/functional (hunk 8 above)
Comment (AlexGuteniev) [reply stl/inc/functional:76]: and below in this file

File: stl/inc/set (hunk 4 above)
Comment (AlexGuteniev) [reply stl/inc/set:53]: it is still not resolved

File: stl/inc/charconv [hunk 9]
```diff
@@ -33,7 +33,6 @@ inline constexpr char _Charconv_digits[] = {'0', '1', '2', '3', '4', '5', '6', '
     'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'};
 static_assert(_STD size(_Charconv_digits) == 36);
 
-// FUNCTION to_chars (INTEGER TO STRING)
```
Comment (SuperWig) [inline stl/inc/charconv:36]: Should the `INTEGER TO STRING` (and other ones) part be kept?

File: stl/inc/functional (hunk 8 above)
Comment (SuperWig) [reply stl/inc/functional:76]: I'll do an audit of `// defined in <>` and replace them with `x, y, z defined in <a>`. Thanks for pointing this out.

File: stl/inc/set [hunk 10]
```diff
@@ -229,10 +225,8 @@ typename set<_Kty, _Keylt, _Alloc>::size_type erase_if(set<_Kty, _Keylt, _Alloc>
 }
 #endif // _HAS_CXX20
 
-// CLASS TEMPLATE multiset
 template <class _Kty, class _Pr = less<_Kty>, class _Alloc = allocator<_Kty>>
 class multiset : public _Tree<_Tset_traits<_Kty, _Pr, _Alloc, true>> {
-    // ordered red-black tree of key values, non-unique keys
```
Comment (fsb4000) [inline stl/inc/set:235]: I think we shouldn't delete the line

Comment (SuperWig) [reply stl/inc/set:235]: That is accidental damage.

**Comment by SuperWig** [comment]
Urgh, didn't realise that would close. I just wanted to rename my branch :(

**Pull request closed without merging**